### PR TITLE
JS RegExp examples (with caveats)

### PR DESCRIPTION
**Pull request by maboa**
I don't necessarily think we should merge this, but I put it out there for discussion.

The issues are:

1. `regexp-prototype-source.html` - specifically:

```
console.log(new RegExp('\n').source === '\\n');
// expected output: true
```

Which I believe should return `true` (source : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/source)

It returns `true` in Firefox and Safari but `false` in Chrome and Opera .

2. `regexp-prototype-tostring.html` - (possibly related to 1.) specifically:

```
console.log(new RegExp('\n', 'g').toString());
// expected output: "/\n/g"
```

This produces the expected output in Firefox and Safari but in Chrome and Opera produces:

```
"/
/g"
```

### CODE REVIEW

**Comment by wbamberg** [comment]
I can reproduce the behavior you're describing with the Chrome console.

I think perhaps this is a genuine incompatibility. In the [browser compat table for this API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/source#Browser_compatibility) there's a row labeled "Escaping" that presumably describes support for this [feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/source#Empty_regular_expressions_and_escaping).  It gets a "?" for Chrome and Opera, I wonder if that should be a "No".

@Elchi3 , do you know anything about this one?

File: live-examples/js-examples/regexp-prototype-source.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<pre>
+<code id="static-js">var regex1 = /fooBar/ig;
+
+console.log(regex1.source);
+// expected output: "foobar"
```
Comment (wbamberg) [inline live-examples/js-examples/regexp-prototype-source.html]: "fooBar"

File: live-examples/js-examples/regexp-prototype-tostring.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<pre>
+<code id="static-js">console.log(new RegExp('a+b+c'));
+// expected output: a+b+c
```
Comment (wbamberg) [inline live-examples/js-examples/regexp-prototype-tostring.html]: I'm getting: `/a+b+c/` here (on Firefox)

File: live-examples/js-examples/regexp-prototype-source.html (hunk 1 above)
Comment (maboa) [reply live-examples/js-examples/regexp-prototype-source.html]: Yes.

File: live-examples/js-examples/regexp-prototype-tostring.html (hunk 2 above)
Comment (maboa) [reply live-examples/js-examples/regexp-prototype-tostring.html]: Yup.

**Pull request merged**